### ISSUE TITLE
Android Authority feed

### ISSUE THREAD

**Issue by PhilC813** [jocmp/capyreader#614]
### Feed URL

https://www.androidauthority.com/feed/

### Add any details, links, or screenshots about the article layout that's missing or wrong

In the following article, the name of the different sections, which are headers ("h2" elements), are stripped out in full content mode.

Article:
https://www.androidauthority.com/new-android-apps-658839/

![2024_12_17_02 38 54](https://github.com/user-attachments/assets/64f9e830-bff0-4d5e-877e-33c1a3e80d20)

Here's the stripped out HTML, which is rather simple:
`<h2 id="[number]">[Header]</h2>`

I would assume that the same would occur for other articles in the feed.

Unless I'm mistaken, header elements shouldn't be stripped out regardless of the feed; they are usually relevant to the content.

Thank you!!

**Comment by PhilC813** [comment]
I also noticed that the embedded YouTube videos are stripped out, but unlike the headers, they seem oddly integrated in the code as they only appear in the JSON data. I understand if this can't be improved in the parser.

"TED Tumblewords" app [section](https://www.androidauthority.com/new-android-apps-658839/#3):
`{"resource":"nc-embed-youtube","video":{"youtubeId":"1Z9fVc6v2aY"}}`

"Carrion" app [section](https://www.androidauthority.com/new-android-apps-658839/#6):
`{"resource":"nc-embed-youtube","video":{"youtubeId":"M6NOM2-UZdw"}}`

**Comment by jocmp** [comment]
Thanks for all the details. The missing headers should be simple. I'll see what I can do for that YouTube video JSON.

**Comment by jocmp** [comment]
I have a custom parser I'll test a little more before adding to the next release (https://github.com/jocmp/mercury-parser/pull/27). 

I haven't found a way to grab the YouTube videos - it looks like they're doing something weird with JavaScript for non-browser clients (like Capy). That said, headers are working. Stay tuned!

<table>
<tr>
<th>Before</th>
<th>After</th>
</th>
</tr>
<tr>
<td valign="top">
<img src="https://github.com/user-attachments/assets/ce2edca1-aecf-450f-8a96-d47220275301" width="300px">
</td>
<td valign="top">
<img src="https://github.com/user-attachments/assets/e27a782a-0df2-4976-850c-3bf7b01249c6" width="300px">
</td>
</tr>
</table>

**Comment by PhilC813** [comment]
Sounds promising! Thank you!

If you made a custom parser for this feed specifically, is it because you don't consider header elements to be relevant to the content for most feeds?

**Comment by jocmp** [comment]
I think headers are important. I want to avoid breaking the core parser since I don't fully understand all the different parts yet. So adding a custom fix is easiest now.

In general sites can misuse headers which is why the previous maintainers built the parser that way. Luckily they [left a note](https://github.com/jocmp/mercury-parser/blob/1be104e6c4ac06dee971356817f2262b10e61d51/src/utils/dom/clean-headers.js#L9-L12) on why the built it that way.

> Remove any headers that appear before all other p tags in the
> document. This probably means that it was part of the title, a
> subtitle or something else extraneous like a datestamp or byline,
> all of which should be handled by other metadata handling.

**Comment by jocmp** [comment]
Updated as of [2024.12.1085-dev](https://github.com/jocmp/capyreader/releases/tag/2024.12.1085-dev)

**Comment by PhilC813** [comment]
So far, articles seem rendered properly with the update to the parser! Thank you so much! It's a feed that I only check occasionally however so I'll let you know if I notice other things down the line.

------

One small thing, albeit not important;
For polls like in [this article](https://www.androidauthority.com/how-google-android-xr-smart-glasses-work-3508597/), the poll and voting choices get rendered in Capy but naturally, can't be interacted with because it must lack the JavaScript.

| Website | Capy |
|--------|--------|
| ![screenshot17362337972574627274782526763858](https://github.com/user-attachments/assets/57cfc4e7-a4d8-4a98-9325-3863f7087d53) | ![Screenshot_20250107_021035](https://github.com/user-attachments/assets/2cb9ae16-940c-42cd-b1f6-406d7edeca17) |

Here's the HTML for that section:
`<div class="e_e"><div class="e_Oh e_Kp"><h3 class="e_Np">Which Android XR form factor are you most excited about?</h3><div class="e_Op">1343 votes</div><div class="e_Pp e_Qh"><button type="button" class="e_Qp"><div class="e_Rp">The headset! Give me that immersion!</div><div class="e_Mp">10<!-- -->%</div><div class="e_Sp"><div class="e_Lp" style="width:10%"></div></div></button><button type="button" class="e_Qp"><div class="e_Rp">The glasses! I want that lightweight design I can use everywhere.</div><div class="e_Mp">46<!-- -->%</div><div class="e_Sp"><div class="e_Lp" style="width:46%"></div></div></button><button type="button" class="e_Qp"><div class="e_Rp">Both.</div><div class="e_Mp">14<!-- -->%</div><div class="e_Sp"><div class="e_Lp" style="width:14%"></div></div></button><button type="button" class="e_Qp"><div class="e_Rp">None. I&#x27;m already invested in another XR/VR/AR platform.</div><div class="e_Mp">3<!-- -->%</div><div class="e_Sp"><div class="e_Lp" style="width:3%"></div></div></button><button type="button" class="e_Qp"><div class="e_Rp">I don&#x27;t care about XR, AR, or VR.</div><div class="e_Mp">28<!-- -->%</div><div class="e_Sp"><div class="e_Lp" style="width:28%"></div></div></button></div></div></div>`

No special element or meaningful class names to identify these polls in the custom parser, but I've compared with their other articles with polls and it appears that the classes `e_Oh` and `e_Kp` are the ones designed for polls, so maybe these could be stripped out?

**Comment by jocmp** [comment]
@PhilC813 good find. Let me see if they're using JS for these. It would be nice to allow this functionality. If nothing else, I'll remove it to avoid the jarring markup.

- https://github.com/jocmp/capyreader/issues/699

**Comment by PhilC813** [comment]
In this article:
https://www.androidauthority.com/best-android-apps-2024-3506812/

The title of the apps in display are stripped out in Capy. Could this be improved?

There's also another example of a poll when you'll need to confirm your results with #699!

**Comment by jocmp** [comment]
yep! I'll roll that in with the polls since the also use `h3` headings. (https://github.com/jocmp/mercury-parser/pull/41)

Here's a preview from my markup tester. "Google Gemini" and "Mozilla Thunderbird" were previously hidden.

<img src="https://github.com/user-attachments/assets/765b7982-5ed6-43fc-8a4d-5b5521099283" width="300px">

**Comment by jocmp** [comment]
@PhilC813 the heading and poll updates are available as of [2025.01.1096-dev](https://github.com/jocmp/capyreader/releases/tag/2025.01.1096-dev)!

**Comment by PhilC813** [comment]
Not sure if it's the result of #699, but now none of the articles in the feed show anything below the header image when in full content mode, as you can see below.

![Image](https://github.com/user-attachments/assets/8876e538-3781-4bdc-b666-6e4bec6bdc60)

**Comment by jocmp** [comment]
@PhilC813 unfortunately it looks like they changed their selectors again. I'm going to need to find a better way than using classes since they're unreliable for Android Authority.

https://github.com/jocmp/mercury-parser/blob/4e98d18a07eab2bd7030308ef368a83d0f5a06c0/src/extractors/custom/www.androidauthority.com/index.js#L23

**Comment by jocmp** [comment]
Tracking via https://github.com/jocmp/capyreader/issues/779

**Comment by PhilC813** [comment]
> unfortunately it looks like they changed their selectors again. I'm going to need to find a better way than using classes since they're unreliable for Android Authority.

Almost like if they want to annoy RSS reader devs. Thank you!

**Comment by jocmp** [comment]
Good callout - this is something strange with [svg](https://developer.mozilla.org/en-US/docs/Web/SVG) pictures. For some reason they're taking up more space than necessary. I may be able to strip them to avoid messing with custom styling.
